server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -627,9 +627,11 @@ public VirtConsumerMap getHostConsumersMap(Owner owner, Iterable<String> hypervi
             .createAlias("owner", "o")
             .createAlias("hypervisorId", "hvsr")
             .add(Restrictions.eq("o.key", ownerKey))
-            .add(this.getHypervisorIdRestriction(hypervisorIds));
+            .add(this.getHypervisorIdRestriction(hypervisorIds))
+            .addOrder(Order.asc("hvsr.hypervisorId"));
 
-        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria);
+        return this.cpQueryFactory.<Consumer>buildQuery(this.currentSession(), criteria)
+            .setLockMode(LockModeType.PESSIMISTIC_WRITE);
     }
 
     private Criterion getHypervisorIdRestriction(Iterable<String> hypervisorIds) {
```

Comment:
I know this isn't your code, but this is painfully inefficient here. We're building a list of restrictions which we then convert into an array later (since in the one place it'd be optimal to do so, we *didn't* use an ArrayList), when the Disjunction object itself has this functionality built in which wouldn't require such shenanigans.

```
Disjunction disjunction = Restrictions.disjunction();
for (String hid : hypervisorIds) {
    disjunction.add(Requestions.eq("hsvr.hypervisorId", hid.toLowerCase()));
}

return disjunction;
```